Userland/Applications/FuzzySearch/FuzzySearchAlgorithms.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2021, Fabian Blatz <fabianblatz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "FuzzySearchAlgorithms.h"
+#include <string.h>
+
+static constexpr int32_t score_match = 16;
+static constexpr int32_t score_gap_start = -3;
+static constexpr int32_t score_gat_extension = -1;
+
+static constexpr int32_t bonus_boundry = score_match / 2;
+static constexpr int32_t bonus_non_word = score_match / 2;
+static constexpr int32_t bonus_camel123 = bonus_boundry + score_gat_extension;
+static constexpr int32_t bonus_consequtive = -(score_gap_start + score_gat_extension);
+static constexpr int32_t bonus_first_char_multiplier = 2;
+
+double FuzzySearchAlgorithms::levenshtein_distance(const StringView& text, const StringView& pattern, const SearchOptions& options)
+{
+    const size_t m = text.length();
+    const size_t n = pattern.length();
+    if (n == 0)
+        return m;
+    if (m == 0)
+        return n;
+
+    size_t costs[n + 1];
+    memset(&costs, 0, n + 1);
```

Comment:
Looks like this doesn't zero the whole costs array.

Reply:
Yep, did not take the type into account, changed it to a `sizeof` usage.